src/aitsi/m3spin/pkb/impl/ParentImpl.java

```diff
@@ -0,0 +1,49 @@
+package aitsi.m3spin.pkb.impl;
+
+import aitsi.m3spin.commons.interfaces.TNode;
+import aitsi.m3spin.pkb.interfaces.Parent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParentImpl implements Parent {
+
+    @Override
+    public List<TNode> getParentedBy(TNode p) {
+
+        TNode n = p;
+        List<TNode> childList = new ArrayList<TNode>();
+
+        while(n.getChild() != null)
+        {
+            n = n.getChild();
```

Comment:
Halo halo, tutaj pętla iteruje po kolejnych pokoleniach - najpierw do listy dodaje dziecko, potem wnuka, prawnuka itd., a powinna iterować bo dzieciach jednego rodzica (podanego jako argument `p`). Skoro robimy po metodą rodzeństwie, to wystarczy dodać do listy "jedyne" dziecko, a potem przechodzić po jego rodzeństwie w prawo. Dobrze myślę?

Reply:
Później na kibelku do podobnych wniosków doszedłem - fixed